index.js

```diff
@@ -0,0 +1,89 @@
+module.exports = {
+	root: true,
+	env: {
+		browser: true,
+		commonjs: true,
+		es6: true,
+		node: true,
+	},
+	globals: {
+		t: true,
+		n: true,
+		OC: true,
+		OCA: true,
+		Vue: true,
+		VueRouter: true
+	},
+	parserOptions: {
+		parser: 'babel-eslint',
+		ecmaVersion: 6
+	},
+	extends: [
+		'eslint:recommended',
+		'plugin:import/errors',
+		'plugin:import/warnings',
+		'plugin:node/recommended',
+		'plugin:vue/essential',
+		'plugin:vue/recommended',
+		'plugin:nextcloud/recommended',
+		'standard'
+	],
+	plugins: ['vue', 'node'],
+	rules: {
+		// space before function ()
+		'space-before-function-paren': ['error', 'never'],
+		// curly braces always space
+		'object-curly-spacing': ['error', 'always'],
+		// stay consistent with array brackets
+		'array-bracket-newline': ['error', 'consistent'],
+		// 1tbs brace style
+		'brace-style': 'error',
+		// tabs only
+		indent: ['error', 'tab'],
+		'no-tabs': ['off'],
+		'vue/html-indent': ['error', 'tab'],
+		// only debug console
+		'no-console': ['error', { allow: ['error', 'warn', 'info', 'debug'] }],
+		// classes blocks
+		'padded-blocks': ['error', { classes: 'always' }],
+		// always have the operator in front
+		'operator-linebreak': ['error', 'before'],
+		// ternary on multiline
+		'multiline-ternary': ['error', 'always-multiline'],
+		// force proper JSDocs
+		'valid-jsdoc': [2, {
+			prefer: {
+				return: 'returns'
+			},
+			requireReturn: false,
+			requireReturnDescription: false
+		}],
+		// es6 import/export and require
+		'node/no-unpublished-require': ['off'],
+		'node/no-unsupported-features/es-syntax': ['off'],
+		// PascalCase components names for vuejs
+		// https://vuejs.org/v2/style-guide/#Single-file-component-filename-casing-strongly-recommended
+		'vue/component-name-in-template-casing': ['error', 'PascalCase'],
+		// force name
+		'vue/match-component-file-name': ['error', {
+			extensions: ['jsx', 'vue', 'js'],
+			shouldMatchCase: true
+		}],
+		// space before self-closing elements
+		'vue/html-closing-bracket-spacing': 'error',
+		// no ending html tag on a new line
+		'vue/html-closing-bracket-newline': ['error', { multiline: 'never' }],
+		// check vue files too
+		'node/no-missing-import': ['error', {
+			tryExtensions: ['.js', '.vue']
+		}],
+		// code spacing with attributes
+		'vue/max-attributes-per-line': ['error', {
```

Comment:
Shall we move to ? Some of us like having everything into one line! @nextcloud/vue

Reply:
We could mark it as `suggestion`, not as `error`

Reply:
So suggestion as one per line if multiple, and max 3 per line in single line?

Reply:
I personally prefer one-per-line a lot, but i don't want to force that onto everyone.

We can agree that we want to prevent lines with 10 properties, so we should simply recommend not using more than 3 per line.

Reply:
Looking at mail as well, Christoph seems to be using something like 3 inline but 1 per line if more than one line.
I'd be fine with that :)
It's pretty clear!

Reply:
@korelstar, @nickvergessen, @tcitworld, @ma12-co, @juliushaertl, @splitt3r, @raimund-schluessler thoughts?

Ok behavior: 
```vue
<ActionCheckbox v-else
	:checked.sync="canUpdate"
	:disabled="saving"
	@change="queueUpdate('permissions')">
	{{ t('files_sharing', 'Allow editing') }}
</ActionCheckbox>
<ActionButton icon="icon-close" @click.prevent.stop="onCancel">
	{{ t('files_sharing', 'Cancel') }}
</ActionButton>
```

bad behavior
```vue
<multiselect v-model="newUser.groups" :options="canAddGroups" :disabled="loading.groups||loading.all"
	tag-placeholder="create" :placeholder="t('settings', 'Add user in group')"
	label="name" track-by="id" class="multiselect-vue"
	:multiple="true" :taggable="true" :close-on-select="false"
	:tag-width="60" @tag="createGroup">
```

Reply:
Personally I'd like this even more:

```
<ActionCheckbox 
	v-else
	:checked.sync="canUpdate"
	:disabled="saving"
	@change="queueUpdate('permissions')"
	>
	{{ t('files_sharing', 'Allow editing') }}
</ActionCheckbox>
```